### PR TITLE
Adde support for Keycloak and collection signing

### PR DESCRIPTION
**Pull request by ironfroggy**

### CODE REVIEW

File: galaxykit/client.py [hunk 1]
```diff
@@ -38,22 +54,46 @@ def __init__(
...
+                    "User-Agent": user_agent(),
+                    "Content-Type": "application/x-www-form-urlencoded",
+                }
+                resp = requests.post(
+                    self.auth_url,
+                    data=payload,
+                    verify=self.https_verify,
+                    headers=headers,
+                )
+                # import pdb;pdb.set_trace()
```
Comment (hendersonreed) [inline galaxykit/client.py:86]: should probably remove this stray comment.

Comment (hendersonreed) [reply galaxykit/client.py:86]: also do we still want all the prints below?

File: galaxykit/collections.py [hunk 2]
```diff
@@ -13,6 +13,12 @@
 from .client import GalaxyClientError
 
 
+def collection_info(client, repository, namespace, collection_name, version):
+    url = f"{client.galaxy_root}content/{repository}/v3/collections/{namespace}/{collection_name}/versions/{version}/"
+    # assert 0, url
```
Comment (hendersonreed) [inline galaxykit/collections.py:18]: another stray debugging line that's been commented out?

File: galaxykit/collections.py [hunk 3]
```diff
@@ -13,6 +13,12 @@
 from .client import GalaxyClientError
 
 
+def collection_info(client, repository, namespace, collection_name, version):
+    url = f"{client.galaxy_root}content/{repository}/v3/collections/{namespace}/{collection_name}/versions/{version}/"
```
Comment (hendersonreed) [inline galaxykit/collections.py:17]: not sure that your url needs the leading `{client.galaxy_root}` - the `client.get()` function prepends the galaxy_root to the passed url (see here: https://github.com/ansible/galaxykit/blob/main/galaxykit/client.py#L91)

File: galaxykit/collections.py [hunk 4]
```diff
@@ -180,4 +187,14 @@ def delete_collection(client, namespace, collection, version):
 
     return client.delete(delete_url, parse_json=False)
  
-    
\ No newline at end of file
+    
+def collection_sign(client, repository, namespace, collection, version, signing_service="ansible-default"):
+    url = f"{client.galaxy_root}content/{repository}/v3/sign/collections/"
```
Comment (hendersonreed) [inline galaxykit/collections.py:192]: same here, don't think we need the leading `{client.galaxy_root}`

File: galaxykit/command.py [hunk 5]
```diff
@@ -292,7 +332,9 @@ def main():
                 (url,) = args.rest
                 print(json.dumps(client.get(url)))
             elif args.operation == "post":
-                raise NotImplementedError
+                url = args.rest[0]
+                body = ""# sys.stdin.read()
```
Comment (hendersonreed) [inline galaxykit/command.py:336]: maybe in the future we can have this take a file or something? not sure the best/most flexible way to take in the input for a POST request but it is probably something that would be handy to have.

**Review by hendersonreed — CHANGES_REQUESTED** [review]
Lots of good changes here! I found a few stray comments and posted a question or two but on the whole this is ready to merge.

File: galaxykit/client.py (hunk 1 above)
Comment (ironfroggy) [reply galaxykit/client.py:86]: 👍

**Review by hendersonreed — APPROVED** [review]

**Review by hendersonreed — APPROVED** [review]

**Pull request merged**